### PR TITLE
Bike feature

### PR DESCRIPTION
**Pull request by ginmrt**
Resolves https://github.com/busbud/stairs/issues/4

Works pretty much like the Usage section of the issue description

### CODE REVIEW

File: bike.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+const Table = require('cli-table2')
+
+
+async function _getUserBikedToWorkDistance(state, userId) {
+  const rows = await state.db.query('SELECT * FROM users WHERE id = $1', [userId + ''])
```
Comment (reveza) [inline bike.js:5]: What is this part used for `[userId + ' ']` ?

File: bike.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  await state.db.query('UPDATE users SET biked_to_work_distance = $1 WHERE id = $2', [distanceMeters, userId + ''])
+  message.raw.tags.push('#bike-distance-saved')
+}
+
+async function _hasUserBikedToWorkBefore(state, userId) {
+  const rows = await state.db.query('SELECT COUNT(*) FROM bike_runs WHERE user_id = $1', [userId + ''])
+  return parseInt(rows[0].count, 10) > 0
+}
+
+async function _handleDistanceInputAbsentForBikeDone(message, state, hasBikedToWorkBefore) {
```
Comment (reveza) [inline bike.js:22]: Why do you call this function `Absent` and the function below `Present`? Could you explain the logic behind the choice of names please :)

File: bike.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    if (achievement) {
+      await message.send(`@team, you just reached the ${achievement.name} (${achievement.location}), with a total of ${achievement.height / 1000} km!`)
+    }
+  }
+}
+
+async function onBikeDone (message, state) {
+  await state.db.query('INSERT INTO users (id, name) VALUES ($1, $2) ON CONFLICT (id) DO UPDATE SET name = $2', [message.author.id, message.author.name])
+
+  const number = message.words[message.words.indexOf(message.doneHash) + 1]
```
Comment (reveza) [inline bike.js:79]: What does `indexOf(message.doneHash) + 1` number represent?

File: bike.js (hunk 1 above)
Comment (ginmrt) [reply bike.js:5]: `userId` is a number here but a text field in the DB. Query didn't work. Doing `variable + ''` is a JS trick to convert the value to a string, and with that change the query works.

File: bike.js (hunk 2 above)
Comment (ginmrt) [reply bike.js:22]: The "I biked to work" signal from the chat app can take some different formats, like:
- `#btw`
- `#btw 5`
- `#btw 2 #save`

where 5 and 2 represent the number of kilometers biked, but the value is not required. There are cases to handle when this distance input is present and cases to handle when it is absent, hence, the naming.

File: bike.js (hunk 3 above)
Comment (ginmrt) [reply bike.js:79]: `indexOf(message.doneHash)` : Position of the hashtag word (e.g. `#btw`) in the array of words of the user message
`indexOf(message.doneHash) + 1`: Position of the optional word following the hashtag word, which we expect to be the number of kilometers biked when present.

See [here](https://github.com/busbud/stairs/pull/6/files#diff-168726dbe96b3ce427e7fedce31bb0bcR52)

**Pull request merged**